resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientResponse.java

```diff
@@ -129,7 +130,24 @@ protected InputStream getEntityStream()
    {
       if (bufferedEntity != null) return new ByteArrayInputStream(bufferedEntity);
       if (isClosed()) throw new ProcessingException(Messages.MESSAGES.streamIsClosed());
-      return getInputStream();
+      InputStream is = getInputStream();
+      return is != null ? new InputStreamWrapper(is, this) : null;
+   }
+   
+   private static class InputStreamWrapper extends FilterInputStream {
+      
+      private ClientResponse response;
+      
+      protected InputStreamWrapper(InputStream in, ClientResponse response) {
+         super(in);
+         this.response = response;
+      }
+      
+      @Override
+      public void close() throws IOException {
+         super.close();
+         this.response.close();

```

Comment:
For the records, I've been discussing this a bit with Jim and we eventually agreed that it should be fine to leave the fix as is.